### DISCUSSION TITLE
RFC: Hide previous marks behavior with more than one drawing task

### DISCUSSION THREAD

**Discussion by srallen** [zooniverse/front-end-monorepo#2439] [General]
**Package** lib-classifier

**Describe the bug**

Found while testing #2001: https://github.com/zooniverse/front-end-monorepo/pull/2001#issuecomment-766685933

Hide/Show previous marks is broken on a workflow with two drawing steps.

1. Hide previous marks on the second step.
2. Press Back.
3. Previous marks are rendered again for the first step.
4. Press Next.
5. Previous marks are hidden again, and can be revealed with the Show Previous Marks button.

**Expected behavior**

**Additional context**

**Comment by srallen** [comment]
@eatyourgreens could you share what project you used that demonstrates this bug?

**Comment by eatyourgreens** [comment]
Sure. It's my test project on staging, with the 'monorepo drawing tools' workflow: project 908, workflow: 3370
http://local.zooniverse.org:3000/projects/eatyourgreens/-project-testing-ground/classify/workflow/3370

**Comment by eatyourgreens** [comment]
Here's a real project that demonstrates the same behaviour. https://frontend.preview.zooniverse.org/projects/lbeiermann/worlds-of-wonder/classify/workflow/8871

**Comment by srallen** [comment]
I don't understand the report. It seems to be functioning as intended with the hide previous marks button scoped in functionality to each task. If this is supposed to work differently, could you elaborate on what the expected behavior should be?

**Comment by eatyourgreens** [comment]
I think Hide Previous Marks persists across the steps of a workflow, but I'm really not sure. If I hide drawn marks, then I'm surprised if they become visible again without me changing the button state. I had a look at a few PRN projects, which use 'Hide Previous Marks' but their workflows all have a single drawing step.

Maybe @beckyrother has UX advice on what we should expect to happen if we hide marks, then go back?

**Comment by eatyourgreens** [comment]
I've set up a workflow so that we can compare with PFE.
https://master.pfe-preview.zooniverse.org/projects/eatyourgreens/-project-testing-ground/classify?reload=0&workflow=3370

The behaviour is weird, after going back from the second step: Hide Previous Marks is checked for 4 marks, but those 4 marks are all visible. Is the checkbox state wrong, or should the marks remain hidden?
![Screenshot of a test subject. Hide Previous Marks is checked for four marks, but those four marks are all shown.](https://user-images.githubusercontent.com/59547/105983258-5dc9de80-6090-11eb-83c7-7ffa2124620f.png)

**Comment by eatyourgreens** [comment]
Also, PFE feels a bit clunky after getting accustomed to that workflow in the new classifier.

**Comment by srallen** [comment]
I'm pretty sure we discussed this during implementation. I'll have to find it and we should document it. If I can find past discussion, I would recommend we leave the behavior as is because I think it's more clear as to what it should be doing at least to me compared to the PFE behavior.

**Comment by srallen** [comment]
I can find the original issue, but not the specific discussion about this; it might have been in slack chat and I'm not going to spend time trying to find it anymore. So, new discussion can happen here. Whatever the result of the discussion is here plus the original issue discussion where we decided on and implemented the variant in function between drawing task and transcription task will go into a belated ADR.
